### PR TITLE
Remove numerical recipes

### PR DESCRIPTION
**Pull request by olebole**
This pull request is a follow-up to a [discussion on the starlink-devel mailing list](https://www.jiscmail.ac.uk/cgi-bin/webadmin?A1=ind1602&L=STARDEV#1).

The Numerical Recipes code is not allowed to redistribute; see http://numerical.recipes/licenses/redistribute.html

Numerical recipes code is used Skycat in `rtd/generic/iqefunc.c` and in cfitsio. This patch replaces `iqefunc.c` by the version from [ESO-MIDAS 15SEPpl1.0](ftp://ftp.eso.org/pub/midaspub/15SEP/sources/15SEPpl1.0.tar.gz) which uses `mpfit.c` instead of `mrqfit.c`. Differently to the solution in ESO-MIDAS, the sort function was replaced by the libc `qsort()`.

Note that the file `astrotcl/cfitsio/eval_y.c` still contains NR code; the whole `astrotcl/cfitsio` subdirectory should be updated with the latest version of cfitsio (which does not contain code from the Numerical Recipes anymore).

### CODE REVIEW

File: rtd/generic/iqefunc.c [hunk 1]
```diff
@@ -150,12 +155,15 @@ float      *bgm;
 float      *bgs;
 int        *nbg;
 {
-  int      n, m, ns, ms, nt, mt;
-  float    *pfb, *pwb, *pf, *pw;
-  float    *pf0, *pf1, *pf2, *pf3, *pfs0, *pfs1, *pfs2, *pfs3;
-  float    *pw0, *pw1, *pw2, *pw3, *pws0, *pws1, *pws2, *pws3;
-  double   val, fks, ba, bm, bs;
-  void     hsort();
+int      n, m, ns, ms, nt, mt;
```
Comment (timj) [inline rtd/generic/iqefunc.c]: Why has the indenting changed here?

File: rtd/generic/iqefunc.c [hunk 2]
```diff
@@ -285,14 +293,20 @@ float      bgv;
...
 int      estm9p();
+
 float    av, dx, dy;
 float    *pf, *pw;
+
 double   val, x, y, dv, xm, ym;
 double   am, ax, ay, axx, ayy, axy;
 
+
+
```
Comment (timj) [inline rtd/generic/iqefunc.c]: blank lines should not be included here.

File: rtd/generic/iqefunc.c [hunk 3]
```diff
@@ -285,14 +293,20 @@ float      bgv;
...
+pw = (float *) 0;
```
Comment (timj) [inline rtd/generic/iqefunc.c]:
```
pw = NULL;
```
instead?

File: rtd/generic/iqefunc.c (hunk 1 above)
Comment (olebole) [inline rtd/generic/iqefunc.c]: I have no idea. I took the file as it is from ESO-MIDAS and changed it only for the qsort() use.

**Comment by timj** [comment]
I'm not entirely sure how much of this is new code from @olebole and how much is lifted directly from MIDAS (hence some of my comments may not be relevant).

@pwdraper would you be able to take a look at this?

File: rtd/generic/iqefunc.c (hunk 3 above)
Comment (olebole) [inline rtd/generic/iqefunc.c]: Similar. I had the choice of a minimal change to the version in starlink, or to the version in MIDAS. I took the second option, which should make the communication with ESO easier.

**Comment by timj** [comment]
It might be clearer if the code that came directly from MIDAS was added to the repo with some kind of ESO author rather than @olebole so it's more obvious?

**Comment by olebole** [comment]
OK, I will create a new pull request, containing three patches: first, the change that come from ESO. Then, my changes, and then the removal of the files.

**Comment by timj** [comment]
Thank you. That would make things much clearer.

**Comment by pwdraper** [comment]
I had panned to, but if there's a clearer patch on the way I'll wait for that.

**Comment by timj** [comment]
@olebole I would have been fine with you just force pushing to this PR branch rather than deleting and starting again. :smile:

**Comment by olebole** [comment]
Sorry, I didn't know how to do that.

**Pull request closed without merging**